google/cloud/storage/client_test.cc

```diff
@@ -221,6 +221,38 @@ TEST_F(ClientTest, LoggingDecoratorsRestClient) {
                           "RetryClient"));
 }
 
+/// @test Verify the constructor creates the right set of RawClient decorations.
+TEST_F(ClientTest, FullStack) {
+  // Reset environment variables, we want the `Options` parameter to control the
+  // structure of the stack.
+  ScopedEnvironment logging("CLOUD_STORAGE_ENABLE_TRACING", absl::nullopt);
+  ScopedEnvironment legacy("GOOGLE_CLOUD_CPP_STORAGE_USE_LEGACY_HTTP",
+                           absl::nullopt);
+
+  // Create a client, use the anonymous credentials because on the CI
+  // environment there may not be other credentials configured.
+  auto options = Options{}
+                     .set<UnifiedCredentialsOption>(MakeInsecureCredentials())
+                     .set<internal::UseRestClientOption>(true)
+                     .set<TracingComponentsOption>({"raw-client"});
+#ifdef GOOGLE_CLOUD_CPP_HAVE_OPENTELEMETRY
```

Comment:
This can be unconditional, no?

Reply:
You know, I carefully went off and read the header before making the comment, and still missed the conditionals.  Sigh.